packages/compass-crud/src/stores/crud-store.js

```diff
@@ -5,8 +5,8 @@ import { findIndex, isEmpty } from 'lodash';
 import StateMixin from 'reflux-state-mixin';
 import HadronDocument from 'hadron-document';
 import util from 'util';
-import createDebug from 'debug';
-const debug = createDebug('mongodb-compass:crud:crud-store');
+import createLogger from '@mongodb-js/compass-logging';
```

Comment:
Since Atlas uses part of this package on the web maybe we might want to pass the logger as a prop to the plugin itself since from what I gather `compass-logging` has a `hadron-ipc` require in it with an `electron` require in that which would probably cause their imports to stop working.
I think at some point we probably want to clean up the interfaces that they're using since it's really only the `Document` Component export from this package. I can create a ticket for that we can do down the line.

Reply:
Well ... we do print a warning, but that's all:

https://github.com/mongodb-js/compass/blob/e4c3083b2f60cd52242e2f8cff15b7ae8cb6fbbe/packages/hadron-ipc/index.js#L11-L18

But you're right, I think the `process.emit()` call in `compass-logging` might throw web usage off? I'll look into it on Monday :)

Reply:
Somehow I totally missed that hah. Sounds good! Have a nice weekend.

Reply:
@Anemy So – updated this to be a bit more restrictive, only loading `hadron-ipc` if we’re inside an electron renderer and made the `process.emit()` call only happen when that method actually exists.

That being said, I’m a little unsure how to test that this actually works in a browser context. I tried to do a clean bootstrap from main and then just loading `packages/compass-crud/lib/index.html` the webpack-generated thing) in a browser, but that already failed with `Uncaught TypeError: Cannot destructure property 'ObjectId' of 'n(...)' as it is undefined.`. Any ideas?

Reply:
@addaleax I don't think have an existing easy way to test it in a browser context. I'll create a ticket for cleaning up those interfaces and being able to test it in a browser context: https://jira.mongodb.org/browse/COMPASS-5114